packages/core/src/messagingProtocols/proteus/ProteusService/CryptoClient/CoreCryptoWrapper/secretKeyGenerator.test.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {Decoder, Encoder} from 'bazinga64';
+
+import {generateSecretKey, CorruptedKeyError} from './secretKeyGenerator';
+
+const systemCryptos = {
+  v0: {
+    encrypt: async (value: Uint8Array) => {
+      return value;
+    },
+    decrypt: async (value: Uint8Array) => {
+      return value;
+    },
+    version: undefined,
+  },
+
+  v01: {
+    encrypt: async (value: Uint8Array) => {
+      return Encoder.toBase64(value).asBytes;
+    },
+    decrypt: async (value: Uint8Array) => {
+      const str = new TextDecoder().decode(value);
+      return Decoder.fromBase64(str).asBytes;
+    },
+    version: undefined,
+  },
+
+  v1: {
+    encrypt: async (value: string) => {
+      const encoder = new TextEncoder();
+      return encoder.encode(value);
+    },
+    decrypt: async (value: Uint8Array) => {
+      const decoder = new TextDecoder();
+      return decoder.decode(value);
+    },
+    version: 1,
+  },
+} as const;
+
+describe('SecretKeyGenerator', () => {
+  beforeEach(async () => {
+    return new Promise(resolve => {
+      Object.values(systemCryptos).forEach(version => {
+        jest.spyOn(version, 'encrypt');
+        jest.spyOn(version, 'decrypt');
+      });
+      const deleteReq = indexedDB.deleteDatabase('test');
```

Comment:
Love the thoroughness of the tests 🚀  One thing: what about moving this `'test'` db name to const variable?